includes/Embed_Block.php

```diff
@@ -211,4 +229,81 @@ protected function render_block_feed( array $attributes ) {
 
 		return (string) ob_get_clean();
 	}
+
+	/**
+	 * Filters whether AMP should be skipped for the current post.
+	 *
+	 * Disables AMP for the <amp-iframe> proxy requests.
+	 *
+	 * @param bool $skip Whether to skip AMP for the current post.
+	 *
+	 * @return bool Whether to skip AMP or not.
+	 */
+	public function skip_amp_for_proxy_request( $skip ) {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( isset( $_GET['_web_story_embed_proxy'] ) && has_block( self::BLOCK_NAME ) ) {
+			return true;
+		}
+
+		return $skip;
+	}
+
+	/**
+	 * Renders the block type output in AMP context.
+	 *
+	 * @param array $attributes Block attributes.
+	 * @param int   $count Counter used for the proxy URL.
+	 *
+	 * @return string Rendered block type output.
+	 */
+	protected function render_block_amp( array $attributes, $count ) {
```

Comment:
In https://github.com/google/web-stories-wp/pull/3260  I took the markup out and put it in it's own file. Worth doing? https://github.com/google/web-stories-wp/pull/3260/files#diff-e0aa2f97d96b0d1e4a8a4fdcf6e4e11cR1

Reply:
Let's first discuss this properly.